### PR TITLE
server_privileges and server_databases modularized 

### PR DESCRIPTION
**Pull request by Piyush3079**
This pr contains the modular code for pages server_databases and server_privileges.
This pr is slight big because some of the core function required for these two files were copied and then imported.
This pr is in continuation of Mod_Js_Config. Before reviewing this pr, review Mod_Js_Config and merge and then compare this pr with merged one.

Before submitting pull request, please check that every commit:

- [ ] Has proper Signed-Off-By
- [ ] Has commit message which describes it
- [ ] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

### CODE REVIEW

File: js/functions.js [hunk 1]
```diff
@@ -5105,3 +5105,61 @@ jQuery.fn.getPostData = function () {
     }
     return dataPost;
 };
+
+/**
+ * Copy of functions copied from different files to make them globally
+ * available so that build does not break during modularisation
+ */
```
Comment (devenbansod) [inline js/functions.js]: If these are to be removed later, please add a note in a TODO stating the same.

File: js/src/ajax.js [hunk 2]
```diff
@@ -0,0 +1,719 @@
+import { PMA_ajaxShowMessage } from './utils/show_ajax_messages';
```
Comment (devenbansod) [inline js/src/ajax.js]: Try to remove these `PMA_` prefixes

File: js/src/consts/doclinks.js [hunk 3]
```diff
@@ -0,0 +1,365 @@
+/**
+ * Definition of links to MySQL documentation.
+ */
+
+export const mysql_doc_keyword = {
+    /* Multi word */
```
Comment (devenbansod) [inline js/src/consts/doclinks.js]: Why is this file needed if everything is commented out?

Also, please avoid underscores in exports. Might want to use PascalCase.

File: js/messages.php [hunk 4]
```diff
@@ -755,6 +755,26 @@ function () {
     Sanitize::printJsValue("PMA_messages['" . $name . "']", $js_message);
 }
```
Comment (devenbansod) [inline js/messages.php]: Why are two files required? js/messages.php and js/src/messages.php

Is it possible to do away with one of them?

File: webpack.config.babel.js [hunk 5]
```diff
@@ -40,10 +40,10 @@ var plugins = [
 export default [{
     mode: MODE,
     entry: {
-        db_search_new: './js/src/db_search.js'
+        index_new: './js/src/index.js'
     },
```
Comment (devenbansod) [inline webpack.config.babel.js]: Why `index_new`?

**Review by devenbansod — CHANGES_REQUESTED** [review]
Leaving a few initial review comments and suggestions.

If you could record it, I'd love to see a working screen video though (to confirm nothing is broken as of now), although I would definitely verify the usage at a later stage before merging.

File: webpack.config.babel.js (hunk 5 above)
Comment (Piyush3079) [reply webpack.config.babel.js]: The `_new.js` suffix helps us to identify that for this file we need to use `src='webhost+webport/js/dist'` and for other `.js` files we need to use `src=js/filename`.
When all the files get revamped we can drop the `_new` suffix and modify `Script.php` for single entrypoint `index.js`.

File: js/messages.php (hunk 4 above)
Comment (Piyush3079) [reply js/messages.php]: I have removed the duplicate files and merged the two into one and changed some of the js files accordingly.

File: js/src/consts/doclinks.js (hunk 3 above)
Comment (Piyush3079) [reply js/src/consts/doclinks.js]: This pr does not contain any of the case changes for variables and removal of `PMA_` prefix as this pr mainly contains the functions going to use in many files.
I was just thinking to create one or more pr once we create and merge some workable code, which will contain changes like using camelCase for variables, PascalCase for exports and dropping `PMA_` prefix.
What do you think about this?

File: js/common_params.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+echo "var common_params = new Array();\n";
+foreach ($header->getJsParams() as $name => $value) {
+    Sanitize::printJsValue("common_params['" . $name . "']", $value);
+}
+
+// echo "var AJAX_Params = new Array();\n";
+// foreach ($header->getDisplay() as $name => $value) {
+//     Sanitize::printJsValue("AJAX_Params['" . $name . "']", $value);
+// }
```
[the remark this answers is not in the document]
Comment (Piyush3079) [reply js/common_params.php]: Removed

File: js/src/utils/show_ajax_messages.js [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+/**
+ * Functions for adding and removing ajax messages
+ */
+
+/**
+ * Module imports
+ */
+import { PMA_Messages as PMA_messages } from '../variables/export_variables';
+// import { PMA_highlightSQL } from './sql';
```
[the remark this answers is not in the document]
Comment (Piyush3079) [reply js/src/utils/show_ajax_messages.js]: Removed

File: libraries/classes/Header.php [hunk 8]
```diff
@@ -164,19 +164,18 @@ public function __construct()
...
-        $this->_scripts->addFile('vendor/jquery/jquery.min.js');
-        $this->_scripts->addFile('vendor/jquery/jquery-migrate.js');
+        $this->_scripts->addFile('vendor/jquery/jquery.min.js');/////////////////////
+        $this->_scripts->addFile('vendor/jquery/jquery-migrate.js');/////////////////
         $this->_scripts->addFile('whitelist.php');
         $this->_scripts->addFile('vendor/sprintf.js');
         $this->_scripts->addFile('ajax.js');
         $this->_scripts->addFile('keyhandler.js');
-        $this->_scripts->addFile('vendor/jquery/jquery-ui.min.js');
+        $this->_scripts->addFile('vendor/jquery/jquery-ui.min.js');/////////////////
```
[the remark this answers is not in the document]
Comment (Piyush3079) [reply libraries/classes/Header.php]: Just checking all these plugins for compatibility with modular code.
Not required now, so I am removing them.

File: js/src/variables/common_params.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
+import { PMA_sprintf } from '../utils/sprintf';
+import { PMA_showCurrentNavigation } from '../functions/navigation';
+/**
+ * Holds common parameters such as server, db, table, etc
+ *
+ * The content for this is normally loaded from Header.php or
+ * Response.php and executed by ajax.js
+ */
+export const PMA_commonParams = (function () {
```
[the remark this answers is not in the document]
Comment (Piyush3079) [reply js/src/variables/common_params.js]: Answered in a comment below :)

File: js/functions.js (hunk 1 above)
Comment (Piyush3079) [reply js/functions.js]: Added a `TODO` in the comment itself.

File: .babelrc [hunk 10]
```diff
@@ -1,3 +1,4 @@
 {
-    "presets": ["env"]
+    "presets": ["env"],
+    "plugins": ["syntax-dynamic-import"]
```
Comment (Piyush3079) [inline .babelrc]: Babel does not support dynamic imports so to compile that, this plugin is added

File: js/src/index.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
+import { AJAX } from './ajax';
```
Comment (Piyush3079) [inline js/src/index.js]: Entry-point for the modular code. Any function or file needs to be loaded at the time of initial loading or only once should be placed or included in this file

File: js/src/server_databases.js [hunk 12]
```diff
@@ -0,0 +1,151 @@
+/* vim: set expandtab sw=4 ts=4 sts=4: */
```
Comment (Piyush3079) [inline js/src/server_databases.js]: This file is copied from the old server_databases.js and then modified accordingly for the modular code.

File: js/src/server_privileges.js [hunk 13]
```diff
@@ -0,0 +1,436 @@
+/* vim: set expandtab sw=4 ts=4 sts=4: */
```
Comment (Piyush3079) [inline js/src/server_privileges.js]: This file is copied from server_privileges.js and then modified accordingly for the modular code.

File: js/src/utils/Sanitise.js [hunk 14]
```diff
@@ -0,0 +1,35 @@
+/**
```
Comment (Piyush3079) [inline js/src/utils/Sanitise.js]: File containing functions for escaping the html and javascript code strings.

File: js/src/utils/JqueryExtended.js [hunk 15]
```diff
@@ -0,0 +1,198 @@
+import $ from 'jquery';
```
Comment (Piyush3079) [inline js/src/utils/JqueryExtended.js]: Extending jQuery in this file with some plugins and custom functions created for phpMyAdmin

File: js/src/utils/menu_resizer.js [hunk 16]
```diff
@@ -0,0 +1,209 @@
+/* vim: set expandtab sw=4 ts=4 sts=4: */
```
Comment (Piyush3079) [inline js/src/utils/menu_resizer.js]: Menu Resizer functions copied from menu_resizer.js and small changes are made to handle some global variables.

File: js/src/utils/show_ajax_messages.js [hunk 17]
```diff
@@ -0,0 +1,190 @@
+/* vim: set expandtab sw=4 ts=4 sts=4: */
```
Comment (Piyush3079) [inline js/src/utils/show_ajax_messages.js]: File containing PMA_ajaxShowMessage and PMA_ajaxRemoveMessage functions.

File: js/src/utils/sprintf.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+import sprintf from 'sprintf-js';
```
Comment (Piyush3079) [inline js/src/utils/sprintf.js]: Sprintf function modified for the new sprintf library.

File: js/src/variables/common_params.js [hunk 19]
```diff
@@ -0,0 +1,100 @@
+import { PMA_sprintf } from '../utils/sprintf';
```
Comment (Piyush3079) [inline js/src/variables/common_params.js]: PMA_commonParams module for handling common params defined in Header.php

File: js/src/variables/export_variables.js [hunk 20]
```diff
@@ -0,0 +1,10 @@
+import { Variables } from './global_variables';
```
Comment (Piyush3079) [inline js/src/variables/export_variables.js]: Exporting variables from module serialised using global variables.

File: js/src/variables/import_variables.js [hunk 21]
```diff
@@ -0,0 +1,20 @@
+import { Variables } from './global_variables';
```
Comment (Piyush3079) [inline js/src/variables/import_variables.js]: Serialising global variables in the module.

File: package.json [hunk 22]
```diff
@@ -14,20 +14,23 @@
   "dependencies": {
     "codemirror": "5.37.0",
     "jquery": "3.3.1",
-    "jquery-migrate": "3.0.1",
+    "jquery-migrate": "3.0.0",
     "jquery-mousewheel": "3.1.13",
-    "jquery-ui": "1.12.1",
+    "jquery-ui": "^1.12.1",
+    "jquery-ui-bundle": "^1.12.1-migrate",
```
Comment (Piyush3079) [inline package.json]: Jquery-ui does have a bundle so this additional library is added to use jquery-ui with jquery.

File: package.json [hunk 23]
```diff
@@ -14,20 +14,23 @@
   "dependencies": {
     "codemirror": "5.37.0",
     "jquery": "3.3.1",
-    "jquery-migrate": "3.0.1",
+    "jquery-migrate": "3.0.0",
```
Comment (Piyush3079) [inline package.json]: The newer version is not compatible. It is throwing some warnings. So, for now, we can use 3.0.0.

**Review by Piyush3079 — COMMENTED** [review]
General comments for files regarding the changes made.

**Review by devenbansod — APPROVED** [review]
Tested locally. Looks good to me.

Please fix the linter errors. I've restarted the selenium test job, let's see if it turns green.

**Comment by devenbansod** [comment]
@Piyush3079 I am merging this for now given the shortage of time and dependency of next PR at #14466 on this PR. 

But we should take steps to take steps to move these changes fast, while also considering the steps needed to build out a workable release from this (i.e. run out of the box from the release zip export).

The selenium tests can't pass right now, since the Login page is itself broken out of the box (since it requires [additional setup](https://github.com/phpmyadmin/phpmyadmin/issues/14355#issue-328825520) right now to test out this code)

**Pull request merged**